[Fix] 홈/주간통계 화면 미기록 처리 및 알림 화면 이동 #171

librawish808 opened the pull request:
## 🔗 관련 이슈
<!-- 이 PR과 연결된 이슈 번호를 명시해주세요 (예: Close #123) -->
- Close #171 

## 📙 작업 설명
- 서버 미기록 응답(null)이 변경됨에 따라, 홈화면과 주간 통계 화면에 반영하였습니다.
   Dto 필드의 모든 타입을 널러블하게 수정했습니다. 
   누락된 필드를 안전하게 파싱할 수 있도록 기본값을 ? = null로 설정했습니다.
   UI State에서도 null값을 그대로 받을 수 있도록 수정했습니다. 
   StatisticsViewModel에 있던 DTO 변환 로직을 WeeklySummaryUiState의 from() 팩토리 메서드로   
   분리했습니다.
- 홈->알림/ 주간통계->알림화면으로 이동기능 수정
   NameBar의 알림아이콘 클릭시, 알림화면을 호출하고 이동하도록 변경하였습니다
   
## 💬 추가 설명 or 리뷰 포인트 (선택)



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **새로운 기능**
  * AI 요약 필드 추가 및 홈/통계 화면에 읽지 않은 알림 수 표시

* **개선사항**
  * 홈·통계에서 알람 화면 이동 네비게이션 추가
  * 데이터 null 안전성 강화 — 미기록 시 "-" 또는 "미기록" 노출
  * 약 복용 회차 및 다음 복용 시간 표시 개선, 혈당·수면·정신 상태 처리 보완

* **버그 수정/안정성**
  * 알림 아이콘 클릭 대상 조정 및 아이콘 크기 변경, 동적 카운트 반영
  * UI 변환 경로 간소화로 안정성 향상
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

librawish808 committed:
Fix: #171 알림 네비게이션 활성화 홈->알림화면 및 주간통계->알림화면

librawish808 committed:
Refactor: #171 HomeRepositoryImpl에서 HomeUiState 수동 매핑 제거 및 from(res)로 일원화

librawish808 committed:
Refactor: #171 HomeResponseDto 필드 어노테이션 정리 및 서버 스펙에 맞게 nullable로 변경

librawish808 committed:
Refactor: #171 HomeScreen 읽지않은 알림 개수 연동 및 null-safe 처리

librawish808 committed:
Refactor: #171 HomeScreen 읽지 않은 알림 개수 연동 및 null값 처리

librawish808 committed:
Refactor: #171 HomeScreen의 각 Container null값 처리

librawish808 committed:
Refactor: #171 DTO Nullable 타입 수정

librawish808 committed:
Refactor: #171 HomeScreen 널값 처리

librawish808 committed:
Refactor: #171 주간통계 Dto 널값 처리

librawish808 committed:
Refactor: #171 주간통계 읽지 않은 알림개수 연동, 알림화면 연결 및 널값 처리

librawish808 committed:
Refactor: #171 주간통계 뷰모델의 DTO 변환 로직을 UI State from() 팩토리 메서드로 분리

librawish808 committed:
Refactor: #171 WeeklySummaryUiState 중첩 클래스 변경에 따른 프리뷰 참조 경로 수정

librawish808 committed:
Refactor: #171 WeeklySummaryCard 널값 처리

librawish808 committed:
Refactor: #171 WeeklySummaryUiState 널값 처리

librawish808 committed:
Refactor: #171 WeeklySummaryUtil 널값 처리

librawish808 committed:
Refactor: #171 ViewModel의 UI 전용 상수(EMPTY_HEALTH_MESSAGE) 제거

librawish808 committed:
Chore: #171 린트 검사

librawish808 committed:
Chore: #171 어노테이션 수정

librawish808 committed:
Merge branch 'develop' into fix/home-statistics-empty-data-#171

# Conflicts:
#	app/src/main/java/com/konkuk/medicarecall/data/dto/response/HomeResponseDto.kt
#	app/src/main/java/com/konkuk/medicarecall/ui/feature/home/component/HomeMedicineContainer.kt
#	app/src/main/java/com/konkuk/medicarecall/ui/feature/home/viewmodel/HomeUiState.kt

librawish808 committed:
Chore: #171 린트 검사

ikseong00 commented on app/src/main/java/com/konkuk/medicarecall/data/repositoryimpl/HomeRepositoryImpl.kt:
```diff
@@ -59,28 +50,7 @@ class HomeRepositoryImpl @Inject constructor(
+            HomeUiState.from(res)
         } catch (e: HttpException) {
             Log.e("HomeRepo", "HTTP error fetching home data: ${e.code()}", e)
             HomeUiState.Companion.EMPTY
```
현재 Data 레이어에 있는 Repository 에서 UI 레이어에 있는 UiState를 참조하고 있습니다.
RepositoryImpl 클래스에서는 응답값을 리턴할 때 Ui 레이어에 없는 단순 Dto 를 리턴하거나, 도메인 모델(현재는 없지만)을 리턴하는 게 좋을 것 같습니다.

그 후 Ui 레이어 혹은 매퍼를 통해 변환하는 게 좋을 것 같습니다.
추후 `from` 대신 `toUiState` 함수가 더 좋을 것 같아요!

참고: https://developer.android.com/topic/architecture?hl=ko#data-layer

librawish808 replied:
아하 거꾸로 참조해서 그런거군요...! 수정해보겠습니다 
리뷰 감사합니다!!

ikseong00 commented on app/src/main/java/com/konkuk/medicarecall/ui/feature/home/navigation/HomeNavigation.kt:
```diff
@@ -28,6 +29,7 @@ fun NavGraphBuilder.homeNavGraph(
             navigateToStateMentalDetailScreen = navigateToStateMentalDetailScreen,
             navigateToGlucoseDetailScreen = navigateToGlucoseDetailScreen,
             mainBackStackEntry = backStackEntry,
+            navigateToAlarm = navigateToAlarmScreen,
```
👍

ikseong00 reviewed (approved):
리뷰 확인 부탁드려요!

librawish808 committed:
Refactor: #171
- Repository → UI 역방향 의존 해결
- 서버 API가 날짜를 받지 않으므로 getHomeSummary()의 date 파라미터 삭제

librawish808 committed:
Refactor: #171 RepositoryImpl에서 UIState 생성 제거 및 DTO 반환 구조로 정리
- RepositoryImpl 내부에서 UIState.from(res) 호출 제거
- 불필요한 로그/중복 코드(HomeViewModel) 정리

librawish808 committed:
Refactor: #171 ViewModel이 Repository에서 UIState 대신 DTO를 받도록 수정

- getHomeUiState() 호출 제거 및 getHomeSummary() 사용으로 변경
- 서버가 날짜를 받지 않으므로 today 파라미터 제거
- softRefreshCurrentElder 과거 구조에서 필요했던 임시 refresh 로직 주석처리

librawish808 committed:
Chore: #171 린트 검사

librawish808 committed:
Chore: #171 린트 검사

librawish808 committed:
Merge remote-tracking branch 'origin/fix/home-statistics-empty-data-#171' into fix/home-statistics-empty-data-#171

# Conflicts:
#	app/src/main/java/com/konkuk/medicarecall/ui/feature/home/viewmodel/HomeViewModel.kt

librawish808 committed:
Chore: #171 린트 검사

librawish808 committed:
Refactor: #171 RepositoryImpl에서 null 역참조로 인한 크래시가 발생하지 않도록 ?. 및 orEmpty() 적용

librawish808 commented:
@alswlekk @ProtossManse 
QA일정으로 인해 오늘 오전중으로 머지해야해서요 급한 요청 죄송합니다
코드리뷰 부탁드립니다!🥹

alswlekk commented on app/src/main/java/com/konkuk/medicarecall/ui/feature/home/viewmodel/HomeUiState.kt:
```diff
@@ -42,50 +45,49 @@ data class HomeUiState(
+    val todayTakenCount: Int?,
+    val todayRequiredCount: Int?,
+    val nextDoseTime: String?,
+    val doseStatusList: List<DoseStatusUiState>? = null,
```
👍

librawish808 replied:
감사합니다!

alswlekk reviewed (approved):
LGTM
